core/templates/domain/topics_and_skills_dashboard/TopicsAndSkillsDashboardFilterObjectFactory.ts

```diff
@@ -20,16 +20,21 @@
 import { downgradeInjectable } from '@angular/upgrade/static';
 import { Injectable } from '@angular/core';
 
+import { ETopicPublishedOptions, ETopicSortOptions } from
+  // eslint-disable-next-line max-len
+  'pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.constants';
+
 export class TopicsAndSkillsDashboardFilter {
   category: string;
   keyword: string;
-  sort: string;
-  status: string;
+  sort: ETopicSortOptions;
+  status: ETopicPublishedOptions;
+
   /**
-   * @param {String} category - category to filter
-   * @param {String} keyword - keyword to filter
-   * @param {String} sort - sort by filter value
-   * @param {String} status - status to filter
+   * @param {String} category - category to filter for.
+   * @param {String} keyword - keyword to filter for.
+   * @param {ETopicSortOptions} sort - Enum, allowed sort-by values.
```

Comment:
This is wrong, right? It should represent a specific value, not a list of possible values. So perhaps "One of the values in {{name of the enum}}."

Ditto for status.

Reply:
Yes, that'll be better. Done.